compiler/rustc_error_codes/src/error_codes/E0782.md

```diff
@@ -0,0 +1,17 @@
+Trait objects must include the `dyn` keyword.
```

Comment:
Please follow the template for the error explanations:

```
[short title]

Erroneous code example:

```compile_fail,E782
failing code
```

[explanations]

```
fixed code
```